llvmlite/ir/instructions.py

```diff
@@ -162,7 +162,7 @@ def descr(self, buf):
 class PredictableInstr(Instruction):
 
     def set_weights(self, weights):
-        operands = [MetaDataString(self.module, "branch_weights")]
+        operands = [MetaDataString(self.module, b"branch_weights")]
```

Comment:
There is no real difference between allowing to pass `unicode` to metadata strings and to arrays. The latter is a way to communicate with your compiled program, the former is a way to communicate with various framework around your program. In one case you are assuming that your program should see human-readable text as UTF-8, in another you are assuming that your e.g. debugger or symbolizer should see human-readable text as UTF-8.

Metadata strings contain such things as function names and paths, neither of which are *required* to be UTF-8, or human-readable text for that matter. Indeed, on Linux, and depending on how you write your code, on Windows, there's no guarantee that paths are valid UTF-8.

And I think in both cases it's a reasonable assumption that any `unicode` passed is a human-readable text that should be converted to UTF-8, because if you're not representing human-readable text as UTF-8 then something is wrong with you.